engine/render/src/render/render_script_camera.cpp

```diff
@@ -0,0 +1,314 @@
+        dmScript::PushVector4(L, camera->m_Data.m_Viewport);
+        lua_settable(L, -3);
+        */
+
+        lua_pushstring(L, "main_camera");
+        lua_pushboolean(L, camera->m_IsMainCamera);
+        lua_settable(L, -3);
+
+    #define PUSH_NUMBER(name, param) \
+        lua_pushstring(L, name); \
+        lua_pushnumber(L, camera->m_Data.param); \
+        lua_settable(L, -3);
+
+        PUSH_NUMBER("fov",          m_Fov);
+        PUSH_NUMBER("aspect_ratio", m_AspectRatio);
+        PUSH_NUMBER("near_z",       m_NearZ);
+        PUSH_NUMBER("far_z",        m_FarZ);
+    #undef PUSH_NUMBER
+
+    #define PUSH_BOOL(name, param) \
+        lua_pushstring(L, name); \
+        lua_pushboolean(L, camera->m_Data.param); \
+        lua_settable(L, -3);
+
+        PUSH_BOOL("orthographic_projection", m_OrthographicProjection);
+        PUSH_BOOL("auto_aspect_ratio",       m_AutoAspectRatio);
+    #undef PUSH_BOOL
+
+        return 1;
+    }
+
+    // This is called when getting a property on the camera module.
+    // If the 'MAIN' property is called, we return the URL to the current main camera.
+    // Otherwise, we just return the tble entry.
+    static int RenderScriptCamera_index(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+        const char* key = luaL_checkstring(L, 2);
+        if (strcmp(key, RENDER_SCRIPT_CAMERA_MAIN_PROPERTY_NAME) == 0)
+        {
+            dmScript::PushURL(L, g_RenderScriptCameraModule.m_MainCameraURL);
+        }
+        else
+        {
+            lua_gettable(L, -2);
+        }
+        return 1;
+    }
+
+    // This is called when setting a property on the camera module.
+    // A user can change the main camera property by setting the MAIN property
+    // to a different camera URL. In that case we check that the URL is an actual
+    // valid camera.
+    static int RenderScriptCamera_newindex(lua_State* L)
+    {
+        // Stack:
+        // 1: table
+        // 2: key
+        // 3: value
+        DM_LUA_STACK_CHECK(L, -3); // must return 0, so we need to pop all three items
+        const char* key = luaL_checkstring(L, 2);
+        if (strcmp(key, RENDER_SCRIPT_CAMERA_MAIN_PROPERTY_NAME) == 0)
+        {
+            RenderCamera* camera = CheckRenderCamera(L, 3, g_RenderScriptCameraModule.m_RenderContext);
+            RenderScriptCameraSetMainCamera(camera->m_URL);
+        }
+        lua_rawset(L, -3); // -2 (set table[key] = value)
+        lua_pop(L, 1);     // -1 (pop table)
+        return 0;
+    }
+
+    static const luaL_reg RenderScriptCameraMain_Methods[] =
+    {
+        {0, 0}
+    };
+
+    static const luaL_reg RenderScriptCameraMain_Meta[] =
+    {
+        {"__newindex",  RenderScriptCamera_newindex},
+        {"__index", RenderScriptCamera_index},
+        {0, 0}
+    };
+
+    static const luaL_reg RenderScriptCamera_Methods[] =
+    {
+        {"get_cameras", RenderScriptCamera_GetCameras},
+        {"get_info",    RenderScriptCamera_GetInfo},
```

Comment:
Q: There is currently no "set_info" or similar to modify a camera (outside of properties), and I think perhaps that it's something people expect? If we decide to do that now I think we should move the update logic into the camera.cpp file instead of passing in explicit view and projection matrices.

Reply:
We have https://defold.com/ref/stable/camera/#set_camera message. And now it's a bit confusing and name clash with a function that does not the same what function does. Maybe we should deprecate it and replace with a new message `set_info`?

Reply:
We do not have a single set_info() function in the current APIs, but we do have a bunch of different get_info() functions, such as sys.get_sys_info(), timer.get_info(), render.get_render_target_info() and so on.

I prefer if we either provide functions to set the various properties of the camera (set_view(), set_projection() and so on) OR as component properties which we can then set AND get.

Reply:
I'm with Björn here. I prefer separate functions for better clarity.
And probably, as a first step, component properties, as they can be get/set and animated.

Reply:
Alright, I'll add functions to get the same properties as we already have in the camera (the data is already owned by the renderer now and not the camera component). Setting view and projection is a different story since we evaluate them on every update. So if we want to support setting custom view and projection matrices we need to keep track of the overrides and some way of going "back" to using the update-based matrices. For now I'll skip that however